src/integer64.c

```diff
@@ -156,7 +156,17 @@ SEXP as_integer64_character(SEXP x_, SEXP ret_){
   char * endpointer;
   for(i=0; i<n; i++){
     str = CHAR(STRING_ELT(x_, i)); endpointer = (char *)str; // thanks to Murray Stokely 28.1.2012
-    ret[i] = strtoll(str, &endpointer, 10);
+    int base = 10; // default
+    int sign = 1;  // default 
+    if (str[0]=='-'){
+      sign = -1;
+      str++;
+    }
+    if (str[0]=='0' && (str[1] == 'x' || str[1] == 'X')){
```

Comment:
won't this be undefined behavior for `as.integer64('0')`? I guess 'no' because `str[0]=='0'` guarantees that `str[1]` is either `'\0'` or another string?

Reply:
As you say, this is not undefined behavior. R guarantees the `str` is a pointer to a null-terminated character array. The test of `str[0]` means that the character at `str[1]` is either NUL or some other character, and testing it is valid.

I will extends the tests to verify.